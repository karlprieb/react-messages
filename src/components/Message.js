import React from 'react'
import PropTypes from 'prop-types'
import Avatar from './Avatar'

import './message.css'
import avatarImage from '../assets/avatar.jpg'

const Message = ({ text, owner }) => {
  const avatarRender = () => {
    if (owner === 'theirs') {
      return <Avatar src={avatarImage} />
    }

    return false
  }

  return (
    <li className={`message message--${owner}`}>
      {avatarRender()}
      <span className='message-text'>{text}</span>
    </li>
  )
}

Message.propTypes = {
  text: PropTypes.string.isRequired,
  owner: PropTypes.string.isRequired
}

export default Message
