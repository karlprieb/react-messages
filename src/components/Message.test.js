import React from 'react'
import Message from './Message'

describe('Message component', () => {
  it('should render a message of yours', () => {
    const wrapper = shallow(<Message text='Some message' owner='yours' />)
    expect(wrapper.find('.message').hasClass('message--yours')).toEqual(true)
    expect(wrapper.text()).toEqual('Some message')
    expect(wrapper).toMatchSnapshot()
  })

  it('should render a message of theirs', () => {
    const wrapper = shallow(<Message text='Another message just for test' owner='theirs' />)
    expect(wrapper.find('.message').hasClass('message--theirs')).toEqual(true)
    expect(wrapper.text()).toEqual('<Avatar />Another message just for test')
    expect(wrapper).toMatchSnapshot()
  })
})
