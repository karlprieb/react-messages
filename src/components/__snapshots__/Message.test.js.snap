// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message component should render a message of theirs 1`] = `
<li
  className="message message--theirs"
>
  <Avatar
    src="avatar.jpg"
  />
  <span
    className="message-text"
  >
    Another message just for test
  </span>
</li>
`;

exports[`Message component should render a message of yours 1`] = `
<li
  className="message message--yours"
>
  <span
    className="message-text"
  >
    Some message
  </span>
</li>
`;
